DatadogCore/Sources/Core/Context/BatteryStatusPublisher.swift

```diff
@@ -24,11 +24,11 @@ internal final class BatteryStatusPublisher: ContextValuePublisher {
     /// Creates a battery status publisher from the given device.
     /// 
     /// - Parameters:
-    ///   - device: The `UIDevice` instance. `.current` by default.
     ///   - notificationCenter: The notification center for observing the `UIDevice` battery changes,
+    ///   - device: The `UIDevice` instance. `.current` by default.
     init(
-        device: UIDevice = .current,
-        notificationCenter: NotificationCenter = .default
+        notificationCenter: NotificationCenter,
+        device: UIDevice = .current
```

Comment:
I was just lazy, but I now fixed it ✅ ⛺ 🧽 

However, there are other places where default values are provided on init basis rather than passed as dependency injection. This makes integration testing impossible. I wish we pay more attention to this in general.

Reply:
No problem, I was just curious :)

But now I'm wondering why providing default values make integration testing impossible? 🤔 
I understand that removing the defaults enforces dependency injection, and it's probably a better pattern to follow. Maybe we should add this as a rule somewhere, wdyt?

Reply:
> But now I'm wondering why providing default values make integration testing impossible? 🤔

Sorry, wrong wording 🙂. It is not "impossible" as long as the actual value is injected from upstream no matter the default value. However, this was not the case for all dependencies refined in this PR - I had to make an extra effort to first find these dependencies and then to inject them from the root configuration. Otherwise, my integration tests would not work with assumed environment.

> Maybe we should add this as a rule somewhere, wdyt?

This is good idea 👍. I created [RUM-6977](https://datadoghq.atlassian.net/browse/RUM-6977) to look into possibility of adding linter rule for this.